### PR TITLE
Drag n drop to switch parent in scenes

### LINKED ISSUE #92: Reparent entity
Add functionality to editor to reparent an existing entity, thus avoiding having to remove the entity and readd it.

### PR DESCRIPTION
**Pull request by Sajdkick**
fixes #92

### CODE REVIEW

File: src/Editor/GUI/Editors/SceneEditor.cpp [hunk 1]
```diff
@@ -7,6 +7,19 @@
 
 using namespace GUI;
 
+Entity* draggedEntity = nullptr;
```
Comment (Chainsawkitten) [inline src/Editor/GUI/Editors/SceneEditor.cpp]: Global variables. Ewww. Add as private member variables instead.

File: src/Engine/Entity/Entity.hpp [hunk 2]
```diff
@@ -28,11 +28,33 @@ class Entity {
...
+        */
+        bool SetParent(Entity* newParent);
+
+        /// Check if entity is a child.
+        /**
+        * @param child The entity you want to check if it is a child.
+        * @param deep Should we also check if it's a child of a child?.
+        * @return The new entity.
+        */
+        bool HasChild(Entity* child, bool deep = true);
```
Comment (Chainsawkitten) [inline src/Engine/Entity/Entity.hpp]: Could be const

File: src/Engine/Entity/Entity.hpp [hunk 3]
```diff
@@ -28,11 +28,33 @@ class Entity {
...
+        * @param newParent The entity you want to be the new parent.
+        * @return Did we manage set the new parent?.
+        */
+        bool SetParent(Entity* newParent);
+
+        /// Check if entity is a child.
+        /**
+        * @param child The entity you want to check if it is a child.
+        * @param deep Should we also check if it's a child of a child?.
+        * @return The new entity.
```
Comment (Chainsawkitten) [inline src/Engine/Entity/Entity.hpp]: Incorrect documentation for return value.

File: src/Engine/Entity/Entity.cpp [hunk 4]
```diff
@@ -62,7 +62,7 @@ bool Entity::SetParent(Entity* newParent) {
 
 }
 
-bool Entity::HasChild(Entity* check_child, bool deep) {
+const bool Entity::HasChild(Entity* check_child, bool deep) {
```
Comment (andens) [inline src/Engine/Entity/Entity.cpp]: This const has no effect; bool is always copied anyway. Placing the const after the parameter list (before the brace) will indicate that the function itself is const.

File: src/Engine/Entity/Entity.hpp [hunk 5]
```diff
@@ -28,11 +28,33 @@ class Entity {
...
+        */
+        bool SetParent(Entity* newParent);
+
+        /// Check if entity is a child.
+        /**
+        * @param child The entity you want to check if it is a child.
+        * @param deep True if we want to check if it's a grandchild, false if we do not.
+        * @return True if it has a child, false if it does not.
+        */
+        bool HasChild(Entity* child, bool deep = true) const;
```
Comment (Chainsawkitten) [inline src/Engine/Entity/Entity.hpp]: child Entity* can also be const.

**Comment by NinjaDanz3r** [comment]
Can one of the admins verify this patch?

**Pull request merged**